invenio_query_parser/utils.py

```diff
@@ -68,17 +68,33 @@ def generate_keywords():
         from invenio_base.globals import cfg
         keyword_mapping = cfg['SEARCH_ELASTIC_KEYWORD_MAPPING']
         json_schema_paths = cfg['JSON_SCHEMA_PATHS']
+        elastic_config_paths = cfg['ELASTIC_MAPPINGS_PATHS']
```

Comment:
If we reach this point of code then we are sure we are in the valid context. In order to ensure performance, at this point you can use a global variable to store the result of the whole process and return it, if it was already computed.